plugin/trino-geospatial/src/test/java/io/trino/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -2326,6 +2326,52 @@ private void assertInvalidGeometryJson(String json, String message)
                 .hasMessage(message);
     }
 
+    @Test
+    public void testGeometryJsonConversion()
+    {
+        // empty geometries should return empty
+        // empty geometries are represented by an empty JSON array in GeoJSON
+        assertGeometryToAndFromJson("POINT EMPTY");
+        assertGeometryToAndFromJson("LINESTRING EMPTY");
+        assertGeometryToAndFromJson("POLYGON EMPTY");
+        assertGeometryToAndFromJson("MULTIPOINT EMPTY");
+        assertGeometryToAndFromJson("MULTILINESTRING EMPTY");
+        assertGeometryToAndFromJson("MULTIPOLYGON EMPTY");
+        assertGeometryToAndFromJson("GEOMETRYCOLLECTION EMPTY");
+
+        // valid nonempty geometries should return as is.
+        assertGeometryToAndFromJson("POINT (1 2)");
+        assertGeometryToAndFromJson("MULTIPOINT ((1 2), (3 4))");
+        assertGeometryToAndFromJson("LINESTRING (0 0, 1 2, 3 4)");
+        assertGeometryToAndFromJson("MULTILINESTRING (" +
+                "(1 1, 5 1), " +
+                "(2 4, 4 4))");
+        assertGeometryToAndFromJson("POLYGON (" +
+                "(0 0, 1 0, 1 1, 0 1, 0 0))");
+        assertGeometryToAndFromJson("POLYGON (" +
+                "(0 0, 3 0, 3 3, 0 3, 0 0), " +
+                "(1 1, 1 2, 2 2, 2 1, 1 1))");
+        assertGeometryToAndFromJson("MULTIPOLYGON (" +
+                "((1 1, 3 1, 3 3, 1 3, 1 1)), " +
+                "((2 4, 6 4, 6 6, 2 6, 2 4)))");
+        assertGeometryToAndFromJson("GEOMETRYCOLLECTION (" +
+                "POINT (1 2), " +
+                "LINESTRING (0 0, 1 2, 3 4), " +
+                "POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0)))");
+
+        // invalid geometries should return as is.
+        assertGeometryToAndFromJson("MULTIPOINT ((0 0), (0 1), (1 1), (0 1))");
+        assertGeometryToAndFromJson("LINESTRING (0 0, 0 1, 0 1, 1 1, 1 0, 0 0)");
+        assertGeometryToAndFromJson("LINESTRING (0 0, 1 1, 1 0, 0 1)");
+    }
+
+    private void assertGeometryToAndFromJson(String wkt)
```

Comment:
Rename the other method (`assertGeoToAndFromJson`) to `assertGeographyToAndFromJson` to avoid confusion with this one.